picard/script/__init__.py

```diff
@@ -101,3 +114,237 @@ def enabled_tagger_scripts_texts():
+
+    def get_value(self, setting):
+        """A safe way of getting the value of the specified property setting, because
+        it handles missing properties by returning None rather than raising an exception.
+
+        Args:
+            setting (str): The setting whose value to return.
+
+        Returns:
+            any: The value of the specified setting, or None if the setting was not found.
+        """
+        if hasattr(self, setting):
+            value = getattr(self, setting)
+            if isinstance(value, str):
+                value = value.strip()
+            return value
+        return None
+
+    def update_last_updated(self):
+        """Update the last updated attribute to the current UTC date and time.
+        """
+        self.last_updated = datetime.datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S UTC')
+
+    def update_script_setting(self, **kwargs):
+        """Updates the value of the specified properties.
+
+        Args:
+            **kwargs: Properties to update in the form `property=value`.
+        """
+        self._update_from_dict(kwargs)
+
+    def _update_from_dict(self, settings):
+        """Updates the values of the properties based on the contents of the dictionary provided.
+        Properties in the `settings` dictionary that are not found in the script object are ignored.
+
+        Args:
+            settings (dict): The settings to update.
+        """
+        updated = False
+        for key, value in settings.items():
+            if value is not None:
+                if hasattr(self, key):
+                    if isinstance(value, str):
+                        value = value.strip()
+                    setattr(self, key, value)
+                    if key not in self._last_updated_ignore_list:
+                        updated = True
+        # Don't overwrite `last_updated` with a generated value if a last_updated value has been provided.
+        if updated and 'last_updated' not in settings:
+            self.update_last_updated()
+
+    def copy(self):
+        """Create a copy of the current script object with updated title and last updated attributes.
+        """
+        new_object = deepcopy(self)
+        new_object.update_script_setting(title=self.title + N_(" (Copy)"))
+        new_object._set_new_id()
+        return new_object
+
+    def to_json(self, indented=False):
+        """Converts the properties of the script object to a JSON formatted string.  Note that only property
+        names listed in `JSON_PROPERTIES` will be included in the output.
+
+        Args:
+            indented (bool): Indent the output.
+
+        Returns:
+            str: The properties of the script object formatted as a JSON string.
+        """
+        items = {key: getattr(self, key) for key in dir(self) if key in self.JSON_OUTPUT}
+        if indented:
+            return json.dumps(items, indent=4, sort_keys=True)
+        return json.dumps(items, sort_keys=True)
```

Comment:
Instead of passing `indented` we could just pass `indent` as an integer or `None`, and simplify code to:

```python
def to_json(self, indent=None):
...
    items = {key: getattr(self, key) for key in dir(self) if key in self.JSON_OUTPUT}
    return json.dumps(items, sort_keys=True, indent=indent)
```

And of course change callers to use `indent=4` instead of `indented=True`.